### PR TITLE
Refactor to protobuf standard style (mostly).

### PR DESCRIPTION
**Pull request by kjmcqueen**
## Overview
Refactor to protobuf standard style (mostly).
Narrow the detection of self-referencial objects to TupleExpr.

### Does this PR relate to an RChain JIRA issue? 
https://rchain.atlassian.net/browse/ROS-369

### Complete this checklist before you submit the PR
- [x] This PR contains no more than 200 lines of code, excluding test code.
- [x] This PR meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards).
- [x] You have someone in mind to assign for review. Make this assignment after submitting the PR.

### Notes
Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else.

### CODE REVIEW

File: rosette/src/Export.cc [hunk 1]
```diff
@@ -35,39 +35,31 @@ void defaultObjectHandler( ObjectCodePB::Object * lvob, pOb ob, ObjectCodePB::Ob
...
 
-    exOb->set_objectid(BASE(ob)->objectId);
+    exOb->set_object_id(BASE(ob)->objectId);
 
-    auto id = exportedIds.find(BASE(ob)->objectId);
-    if (id != exportedIds.end()) {
-        // Don't export an object more than once. This effectively eliminates problems
-        // with self referential objects and loops in the object tree.
+    std::string type = BASE(ob)->typestring();
+    auto oh = handlers.find(type);
```
Comment (pyrocto) [inline rosette/src/Export.cc]: Why did you remove the check for whether the object had already been exported?

Comment (kjmcqueen) [reply rosette/src/Export.cc]: It seemed to be too generalized.  There is a use of objects that I don't fully understand wherein they take on different values. This caused some symbol values to not be exported.  I had originally implemented this due to the self-referencial nature of the TupleExpr object causing an endless loop and a crash.  I'm still struggling over the way objects mutate during the mixed compile and execute phases.

Here's an example of a litvec containing a table (that holds a tuple). The elements contain symbols that apparently have the same object id, yet contain different values.

code_block {
  object_id: 2649
  litvec {
    ob {
      type: OT_PROC
      object_id: 2637
      meta_id: 32
      parent_id: 33
      proc {
        env {
          type: OT_TBL_OBJECT
          object_id: 359
          meta_id: 1125
          parent_id: 357
          tbl_object {
            validextent {
              type: OT_FIXNUM
              object_id: 1640
              meta_id: 299
              parent_id: 300
              fixnum {
                value: 504
              }
            }
            keyvec {
              type: OT_TUPLE
              object_id: 1303
              meta_id: 1150
              parent_id: 9
              tuple {
                elements {
                  type: OT_SYMBOL
                  object_id: 1637
                  meta_id: 308
                  parent_id: 309
                  symbol {
                    name: "Timer-SBO"
                  }
                }
                elements {
                  type: OT_SYMBOL
                  object_id: 1637
                  meta_id: 308
                  parent_id: 309
                  symbol {
                    name: "RblStack-SBO"
                  }
                }
                elements {
                  type: OT_SYMBOL
                  object_id: 1637
                  meta_id: 308
                  parent_id: 309
                  symbol {
                    name: "Tuple-SBO"
                  }
                }
                elements {
                  type: OT_SYMBOL
                  object_id: 1637
                  meta_id: 308
                  parent_id: 309
                  symbol {
                    name: "RblTable-SBO"
                  }
                }
                elements {
                  type: OT_SYMBOL
                  object_id: 1637
                  meta_id: 308
                  parent_id: 309
                  symbol {
                    name: "Reader-SBO"
                  }
                }
.
.
.

This may be due to objects being copied without a new object id being created. I'm just not sure how all this works yet.

Comment (pyrocto) [reply rosette/src/Export.cc]: > This may be due to objects being copied without a new object id being created.
That seems most likely to me, and was why I brought up the copy constructor when we talked earlier.

Comment (tymm) [reply rosette/src/Export.cc]: > I had originally implemented this due to the self-referencial nature of the TupleExpr object causing an endless loop and a crash.

Out of curiosity: What are you referring to when you say self-referencial nature of the TupleExpr?

File: rosette/src/Export.cc [hunk 2]
```diff
@@ -200,19 +192,28 @@ void letExprObjectHandler( ObjectCodePB::Object * lvob, pOb ob, ObjectCodePB::Ob
 }
 
 void tupleExprObjectHandler( ObjectCodePB::Object * lvob, pOb ob, ObjectCodePB::ObType pbtype ) {
-    ObjectCodePB::TupleExpr *teob = lvob->mutable_tupleexpr();
+    ObjectCodePB::TupleExpr *teob = lvob->mutable_tuple_expr();
     TupleExpr * te = (TupleExpr *)ob;
 
+    // There are cases where the TupleExpr references itself in the "rest" field. This could
```
Comment (tymm) [inline rosette/src/Export.cc]: Can you tell me where and how you observed this?

Comment (kjmcqueen) [reply rosette/src/Export.cc]: CodeId=1708
litvec:
   0:   {RequestExpr} (1690)
   1:   '**SOURCE-FILE** (1644)
   2:   "./boot.rbl" (1688)
codevec:
   0:   alloc 2
   1:   liti 1,arg[0]
   2:   liti 2,arg[1]
   3:   add 2,rslt
   5:   rtn/nxt

code_block {
  object_id: 1708
  litvec {
    ob {
      type: OT_REQUEST_EXPR
      object_id: 1690
      meta_id: 119
      parent_id: 120
      request_expr {
        target {
          type: OT_SYMBOL
          object_id: 1644
          meta_id: 308
          parent_id: 309
          symbol {
            name: "add"
          }
        }
        msg {
          type: OT_TUPLE_EXPR
          object_id: 1689
          meta_id: 1154
          parent_id: 108
          tuple_expr {
            rest {
              type: OT_TUPLE_EXPR
              object_id: 320
              meta_id: 1154
              parent_id: 108
              tuple_expr {
                rest {
                  object_id: 320
                  looped: true
                }
              }
            }
          }
        }
      }
    }
    ob {
      type: OT_SYMBOL
      object_id: 1644
      meta_id: 308
      parent_id: 309
      symbol {
        name: "**SOURCE-FILE**"
      }
    }
    ob {
      type: OT_RBL_STRING
      object_id: 1688
      meta_id: 23
      parent_id: 24
      rbl_string {
        value: "./boot.rbl"
      }
    }
  }
  codevec {
    opcodes: "\002\004\001\272\022\272\0028\000\243\000M0$nV\254\006\000\000\000\000\000\000"
  }
}


The "looped" field indicates that the "rest" pointer in the TupleExpr points to itself, i.e. the TupleExpr containing it.
